File: platform/hackattic/kata/a_case_of_snakes/a_case_of_snakes.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		input := scanner.Text()

		regex := regexp.MustCompile(`([A-Z]+|\d+)[^A-Z\d]*`)
		words := regex.FindAllString(input, -1)

		if len(words) > 1 {

			for i, word := range words {
				words[i] = strings.ToLower(word)
			}

			result := strings.Join(words, "_")
			fmt.Println(result)
		}

	}
}

File: platform/hackattic/kata/a_case_of_snakes/a_case_of_snakes.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
})

rl.on('line', (input) => {
  const words = input.split(/(?=[A-Z])/);
  const [_, ...rest] = words
  const result =  rest.map(word => word.toLowerCase()).join('_')
  console.log(result)
})